float64 data
---
bool state